frontend/appflowy_flutter/lib/plugins/document/presentation/editor_page.dart

```diff
@@ -153,6 +155,22 @@ class _AppFlowyEditorPageState extends State<AppFlowyEditorPage> {
     final (bool autoFocus, Selection? selection) =
         _computeAutoFocusParameters();
 
+    final isRTL =
+        context.read<AppearanceSettingsCubit>().state.layoutDirection ==
+            LayoutDirection.rtlLayout;
+    final layoutDirection = isRTL ? TextDirection.rtl : TextDirection.ltr;
+
+    // only show the rtl item when the layout direction is ltr.
+    for (final item in textDirectionItems) {
+      if (isRTL) {
+        if (toolbarItems.every((element) => element.id != item.id)) {
+          toolbarItems.add(item);
+        }
+      } else {
+        toolbarItems.removeWhere((element) => element.id == item.id);
+      }
+    }
```

Comment:
Maybe something like:
```suggestion
  _setRTLToolbarItems(isRL);
```

And then a method like this:
```
  void _setRTLToolbarItems(bool isRTL) {
    toolbarItems.removeWhere((item) => item.id.contains('text_direction_'));

    if (isRTL) {
      toolbarItems.addAll(textDirectionItems);
    }
  }
```
Additionally abstract this `'text_direction_'` into a `const textDirectionPrefix = "text_direction_";`

Reply:
And an integration test for this logic

Reply:
I prefer to filter the text direction items with their IDs.

```dart
  void _setRTLToolbarItems(bool isRTL) {
    final textDirectionItemIds = textDirectionItems.map((e) => e.id);
    // clear all the text direction items
    toolbarItems.removeWhere(
      (item) => textDirectionItemIds.contains(item.id),
    );
    // only show the rtl item when the layout direction is ltr.
    if (isRTL) {
      toolbarItems.addAll(textDirectionItems);
    }
  }
```

Reply:
> And an integration test for this logic

Added.